packages/asset-swapper/contracts/src/CoFiXSampler.sol

```diff
@@ -0,0 +1,226 @@
+/*
+
+  Copyright 2019 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.5.9;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-utils/contracts/src/DeploymentConstants.sol";
+import "./interfaces/ICoFiXController.sol";
+import "./interfaces/ICoFiXFactory.sol";
+import "./interfaces/ICoFiXPair.sol";
+import "./interfaces/INestOracle.sol";
+
+
+contract CoFiXSampler is
+    DeploymentConstants,
+    SamplerUtils
+{
+    /// @dev Gas limit for CoFiX calls.
+    uint256 constant private COFIX_CALL_GAS = 450e3; // 450k
+
+    struct OraclePrice {
+        uint256 ethAmount;
+        uint256 erc20Amount;
+        uint256 blockNum;
+        uint256 K;
+        uint256 theta;
+    }
+
+    /// @dev Sample sell quotes from CoFiX.
+    /// @param takerToken Address of the taker token (what to sell).
+    /// @param makerToken Address of the maker token (what to buy).
+    /// @param takerTokenAmounts Taker token sell amount for each sample.
+    /// @return makerTokenAmounts Maker amounts bought at each taker token
+    ///         amount.
+    function sampleSellsFromCoFiX(
+        address takerToken,
+        address makerToken,
+        uint256[] memory takerTokenAmounts
+    )
+        public
+        view
+        returns (uint256[] memory makerTokenAmounts)
+    {
+        // make sure it's not WETH-WETH
+        _assertValidPair(makerToken, takerToken);
+        uint256 numSamples = takerTokenAmounts.length;
+        makerTokenAmounts = new uint256[](numSamples);
+        ICoFiXPair takerTokenPair = takerToken == _getWethAddress() ?
```

Comment:
This appears to be similar to DODO where there is a "base" token (token0) and a quote token (token1). And we need to get the oracle data of the "quote" token (token1). And branch off the selector based on the quote/base.

Thhen maybe we could just do:

```
for (uint256 i = 0; i < numSamples; i++) {
...
                (didSucceed, bytes memory resultData) =
                    address(pool).staticcall.gas(COFIX_CALL_GAS)(
                        abi.encodeWithSelector(
                            takerToken == quoteToken ? ICoFiXPair(0).calcOutToken0.selector : ICoFiXPair(0).calcOutToken1.selector,
                            takerTokenAmounts[i],
                            _getOraclePrice(quoteToken)
                    ));
                makerTokenAmounts[i] = abi.decode(resultData, (uint256[]))[0];
```

I may have got the 0 and 1 around the wrong way though.